rules-tests/CodingStyle/NodeAnalyzer/UseImportNameMatcher/UseImportNameMatcherTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\CodingStyle\NodeAnalyzer\UseImportNameMatcher;
+
+use Iterator;
+use PhpParser\Node\Stmt\Use_;
+use PhpParser\ParserFactory;
+use PHPUnit\Framework\Attributes\DataProvider;
+use Rector\CodingStyle\NodeAnalyzer\UseImportNameMatcher;
+use Rector\Core\PhpParser\Node\BetterNodeFinder;
+use Rector\Naming\Naming\UseImportsResolver;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\Testing\PHPUnit\AbstractLazyTestCase;
+use Rector\Testing\TestingParser\TestingParser;
+use Rector\Tests\CodingStyle\NodeAnalyzer\UseImportNameMatcher\Fixture\FixtureAnnotation\CustomAnnotation;
+
+class UseImportNameMatcherTest extends AbstractLazyTestCase
```

Comment:
I prefer real `rules()` usage instead of creating new test class, as it hard to debug, you can add more test with specific config under `tests/Issues`

https://github.com/rectorphp/rector-src/tree/main/tests/Issues

and add fixture with expected output with config there.

Reply:
Sure I'll need to see how to do what you ask.
I'll take a look to the otheres tests.